### PR TITLE
ols solver

### PR DESCRIPTION
**Pull request by josephsalmon**
I have added an ols solver  (CD + ISTSA / FISTA included), on the datasets: Boston + Simulated

### CODE REVIEW

File: benchmarks/ols/solvers/cd.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    @staticmethod
+    @njit
+    def sparse_cd(X_data, X_indices, X_indptr, y, L, n_iter):
+        n_features = len(X_indptr) - 1
+        w = np.zeros(n_features)
+        R = np.copy(y)
+        for _ in range(n_iter):
+            for j in range(n_features):
+                old = w[j]
```
Comment (mathurinm) [inline benchmarks/ols/solvers/cd.py:70]: @josephsalmon you can add 
```
if L[j] == 0.:
    continue
```
as in the cd Lasso caso

Comment (josephsalmon) [reply benchmarks/ols/solvers/cd.py:74]: thx. done.

**Comment by josephsalmon** [comment]
Anything else needed here @tomMoral?

File: benchmarks/ols/datasets/simulated.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class Dataset(BaseDataset):
+
+    name = "Simulated"
+
+    # List of parameters to generate the datasets. The benchmark will consider
+    # the cross product for each key in the dictionary.
+    parameters = {
+        'n_samples, n_features': [
+            (500, 1000),
+            (200, 5000)]
```
Comment (agramfort) [inline benchmarks/ols/datasets/simulated.py:15]: for OLS I think it's not really relevant to test with n_features > n_sample

Comment (josephsalmon) [reply benchmarks/ols/datasets/simulated.py]: agreed. I switch the values between the two.

**Comment by agramfort** [comment]
can you merge with master and add OLS to readme.rst ?

File: README.md [hunk 3]
```diff
@@ -37,6 +37,10 @@ Use `benchopt run -h` for more details about these options, or visit https://ben
 
 ### List of optimization problems available
 
+- `ols`: aka ordinary least-squares. This consists in solving the following program:
+
+  ```min_w 1 / 2 * ||y - Xw||^2_2```
```
Comment (agramfort) [inline README.md:42]: you version is too old. This file is gone in current master. REvert this commit and do:

git pull origin master 

in this branch

Comment (josephsalmon) [reply README.md]: I hope I did it right. Let me know.

**Comment by agramfort** [comment]
success https://22-227720375-gh.circle-artifacts.com/0/dev/index.html

merging. We can improve later.

File: benchmarks/ols/solvers/cd.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    def run(self, n_iter):
+        filterwarnings("ignore", category=PerformanceWarning)
+
+        if sparse.issparse(self.X):
+            L = sparse.linalg.norm(self.X, axis=0) ** 2
+            self.w = self.sparse_cd(
+                self.X.data, self.X.indices, self.X.indptr, self.y,
+                L, n_iter)
+        else:
+            L = norm(self.X, axis=0) ** 2
```
[the remark this answers is not in the document]
Comment (mathurinm) [reply benchmarks/ols/solvers/cd.py:44]: (x ** 2).sum(axis =0) works for both sparse and dense, too

[the remark this answers is not in the document]
Comment (josephsalmon) [reply benchmarks/ols/solvers/cd.py:44]: I have edited the code according to @mathurinm 's proposition.

[the remark this answers is not in the document]
Comment (tomMoral) [reply benchmarks/ols/solvers/cd.py:44]: Then it is probably a good choice here :)

**Pull request merged**